src/Compilers/CSharp/Portable/Symbols/Source/SourceMemberContainerSymbol.cs

```diff
@@ -2491,8 +2484,11 @@ private static bool ContainsModifier(SyntaxTokenList modifiers, SyntaxKind modif
             {
                 membersByName = membersAndInitializers.NonTypeMembers.ToDictionary(s => s.Name, StringOrdinalComparer.Instance);
 
-                // Merge types into the member dictionary
-                AddNestedTypesToDictionary(membersByName, GetTypeMembersDictionary());
+                if (!IsTupleType)
```

Comment:
>if (!IsTupleType) [](http://example.com/codeflow?start=16&length=17)

The rational behind thios change is not clear.

Reply:
This code was removed from `AddOrWrapTupleMembers`:
```
                        case SymbolKind.NamedType:
                            // We are dropping nested types, if any. Pending real need.
                            break;
```

Reply:
Doing that made sense for the world when TypleTypes were separate symbols. Simply dropping nested types for a symbol representing a type declared in source doesn't make sense in my opinion. One can declare a nested type in source, why do we drop them? Is there a real need?

Reply:
I don't mind changing this. I was just trying to maintain current behavior to avoid extra scope for this PR.